frontend/packages/console-app/src/components/modals/clone/clone-pvc-modal.tsx

```diff
@@ -104,6 +122,20 @@ const ClonePVCModal = withHandlePromise((props: ClonePVCModalProps) => {
     });
   };
 
+  const filterUnits = () => {
+    let filteredDropdown = {};
+    _.each(dropdownUnits, () => {
+      if (defaultSize[1] === 'Mi') {
+        filteredDropdown = { Mi: dropdownUnits.Mi };
+      }
+      if (defaultSize[1] === 'Gi' || defaultSize[1] === 'Mi') {
+        filteredDropdown = { ...filteredDropdown, Gi: dropdownUnits.Gi };
+      }
+    });
+    filteredDropdown = { ...filteredDropdown, Ti: dropdownUnits.Ti };
+    return filteredDropdown;
+  };
```

Comment:
May be you would like to define this at one place instead of recreating the function twice.

Reply:
Do we need to iterate over `dropdownUnits` when the check is on `defaultSize[1]` ?
May be I am missing something :)